samza-core/src/main/java/org/apache/samza/zk/ZkUtils.java

```diff
@@ -324,4 +398,45 @@ public void subscribeToProcessorChange(IZkChildListener listener) {
     zkClient.subscribeChildChanges(keyBuilder.getProcessorsPath(), listener);
     metrics.subscriptions.inc();
   }
+
+  /**
+   * Represents zookeeper processor node.
+   */
+  private static class ProcessorNode {
```

Comment:
I think it is nice to use Data Access Object (DAO) pattern here. However, it still doesn't explain why it should be a completely separate class. I think the intention behind `ProcessorData` was similar.

>  I think ephemeralPath doesn't belong to ProcessorData.

I am not following why ephemeralPath cannot belong to ProcessorData. Can you please explain your thoughts?

> In future when the codebase evolves, will help readability.

Code readability is undeniably important. But it is not helping with readability right now, at least for me. 

> ProcessorNode is a place for storing creationTime, numberOfChildren and other statistics along with ACL related info in future

It's nice to plan for future. These will be good stats and data to have in the local object state. It still doesn't explain why we need 2 different classes. Why can't be just maintain one class `ProcessorNode` or `ProcessorData`?